src/ess/powder/correction.py

```diff
@@ -2,81 +2,135 @@
     """
-    Normalize event data by a monitor.
+    _expect_monitor_covers_range_of_detector(
+        detector=detector, monitor=monitor, dim="wavelength"
+    )
+    norm = broadcast_uncertainties(
+        monitor, prototype=detector, mode=uncertainty_broadcast_mode
+    )
+    return detector.bins / sc.lookup(norm, dim="wavelength")
+
+
+def normalize_by_monitor_integrated(
+    detector: DataWithScatteringCoordinates[RunType],
+    *,
+    monitor: WavelengthMonitor[RunType, CaveMonitor],
+    uncertainty_broadcast_mode: UncertaintyBroadcastMode,
+) -> NormalizedRunData[RunType]:
+    """Normalize detector data by an integrated monitor.
 
-    The input is converted to wavelength if it does not already contain wavelengths.
+    The monitor is integrated according to
+
+    .. math::
+
+        M = \\sum_{i=0}^{N-1}\\, m_i (x_{i+1} - x_i) I(x_i, x_{i+1}),
+
+    where :math:`m_i` is the monitor intensity in bin :math:`i`,
+    :math:`x_i` is the lower bin edge of bin :math:`i`, and
+    :math:`I(x_i, x_{i+1})` selects bins that are within the range of the detector.
 
     Parameters
     ----------
-    data:
-        Input event data.
+    detector:
+        Input event data in wavelength.
     monitor:
-        A histogrammed monitor.
-    wavelength_edges:
-        If given, rebin the monitor with these edges.
-    smooth_args:
-        If given, the monitor histogram is smoothed with
-        :func:`ess.powder.lowpass` before dividing into `data`.
-        `smooth_args` is passed as keyword arguments to
-        :func:`ess.powder.lowpass`. If ``None``, the monitor is not smoothed.
+        A histogrammed monitor in wavelength.
+    uncertainty_broadcast_mode:
+        Choose how uncertainties of the monitor are broadcast to the sample data.
 
     Returns
     -------
     :
-        `data` normalized by a monitor.
+        `detector` normalized by a monitor.
     """
-    if "wavelength" not in monitor.coords:
-        monitor = monitor.transform_coords(
-            "wavelength",
-            graph={**beamline.beamline(scatter=False), **tof.elastic("tof")},
-            keep_inputs=False,
-            keep_intermediate=False,
-            keep_aliases=False,
+    dim = monitor.dim
+    if not monitor.coords.is_edges(dim):
+        raise sc.CoordError(
+            f"Monitor coordinate '{dim}' must be bin-edges to integrate the monitor."
         )
+    _expect_monitor_covers_range_of_detector(
+        detector=detector, monitor=monitor, dim=dim
+    )
+
+    # Clip `monitor` to the range of `detector`.
+    det_coord = (
+        detector.coords[dim] if dim in detector.coords else detector.bins.coords[dim]
+    )
+    lo = det_coord.min()
+    hi = det_coord.max()
+    # hi is shifted towards MINUS infinity because label-based indexing with bin-edges
+    # in inclusive for the upper edge. But we don't want the literal upper edge
+    # of the detector to be included.
+    hi.value = np.nextafter(hi.value, -np.inf)
```

Comment:
This was broken in my first implementation. I used `np.nextafter(hi.value, +np.inf)`. Please check that the logic is sound.

Reply:
Conclusion from offline discussion was: `rebin` actually works.